foxtrot-common/src/main/java/com/flipkart/foxtrot/common/query/numeric/BetweenFilter.java

```diff
@@ -80,6 +76,44 @@ public boolean isFilterTemporal() {
         return validationErrors;
     }
 
+    @Override
+    public int hashCode() {
+        int result = getOperator().hashCode();
+        result = 31 * result + getField().hashCode();
+        if(!getField().equals("_timestamp")) {
+            result = result * 21 + (getFrom() == null
+                                    ? 43
+                                    : getFrom().hashCode());
+            result = result * 23 + (getTo() == null
+                                    ? 29
+                                    : getTo().hashCode());
+        }
+        else {
+            result = result * 21 + Long.valueOf(getFrom().longValue() / (long) 30000).hashCode();
+            result = result * 23 + Long.valueOf(getTo().longValue() / (long) 30000).hashCode();
+        }
+        result = result * 59 + (this.isTemporal()
+                                ? 79
+                                : 97);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if(this == o) {
+            return true;
+        }
+        else if(!(o instanceof BetweenFilter)) {
+            return false;
+        }
+
+        BetweenFilter that = (BetweenFilter) o;
+
+        return getField().equals(that.getField()) && getOperator().equals(that.getOperator()) &&
+                isFilterTemporal() == that.isFilterTemporal() && getFrom().equals(that.getFrom()) &&
+                getTo().equals(that.getTo());
```

Comment:
Logic looks shaky. Please check. Use the cache acceptable flag here to govern the behaviour